main.py

```diff
@@ -85,8 +94,145 @@ def recording_file(rec_file):
     run = subparsers.add_parser("run", description="Run the model on the given data")
     run.add_argument("saved_model", nargs=1, type=existing_model)
     run.add_argument("--output-file", "-o", nargs="?", default=None, type=str)
-    run.add_argument("recordings", nargs="+", type=recording_file)
+    run.add_argument("recording", nargs="+", type=recording_file)
 
     return parser.parse_args()
 
-# process_audio_directory
+
+def get_data_from_dir(data_dir, preprocess_method):
+    """
+    Loads the data from a given data directory, giving back both inputs and labels
+    :param data_dir: The directory to load data from
+    :param preprocess_method: The method to use for preprocessing (mfcc | fft)
+    :return: The inputs and labels from the data directory
+    """
+    inputs = None
+    labels = None
+
+    accent_class_folders = [folder for folder in os.listdir(data_dir)
```

Comment:
I'd take a look at the `glob` and `pathlib.Path` libraries. Both have really nice directory parsing functions to make analyzing directories a lot easier.